packages/cli/src/lib/commands/init/processes/generate-lh-budgets.unit.test.ts

```diff
@@ -0,0 +1,54 @@
+import { handleBudgetsGeneration } from './generate-lh-budgets';
+import { RcJson } from '@push-based/user-flow';
+import deriveHelpers = require('../derive-budgets-from-lhr');
+import fileHelpers = require('../../../core/file');
+
+jest.mock('../derive-budgets-from-lhr');
+jest.mock('../../../core/file');
+
+describe('generate LH budgets', () => {
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  })
+
+  it('should not check if file exist if generateBudgets is false', async () => {
+    const readFileSpy = jest.spyOn(fileHelpers, 'readFile');
+    await handleBudgetsGeneration({ generateBudgets: false })({} as RcJson);
+    expect(readFileSpy).not.toHaveBeenCalled();
+  })
+
+  it('should not create flow if generateBudgets is false', async () => {
+    const writeFileSpy = jest.spyOn(fileHelpers, 'writeFile');
+    await handleBudgetsGeneration({ generateBudgets: false })({} as RcJson);
+    expect(writeFileSpy).not.toHaveBeenCalled();
+  });
+
+  it('should not write a budgets file if generateBudgets is true and file budgets file already exists', async () => {
+    const readFileSpy = jest.spyOn(fileHelpers, 'readFile').mockReturnValue('Not Empty String' as never);
+    const writeFileSpy = jest.spyOn(fileHelpers, 'writeFile');
+    await handleBudgetsGeneration({ generateBudgets: true })({} as RcJson);
+    expect(readFileSpy).toHaveBeenCalled();
+    expect(writeFileSpy).not.toHaveBeenCalled();
+  });
+
+  it('should create default budgets file if generateBudgets is true and lhr is passed', async () => {
+    jest.spyOn(fileHelpers, 'readFile').mockReturnValue('' as never);
+    const deriveBudgetsFromLhrSpy = jest.spyOn(deriveHelpers, 'deriveBudgetsFromLhr');
+    const writeFileSpy = jest.spyOn(fileHelpers, 'writeFile');
+    await handleBudgetsGeneration({ generateBudgets: true })({} as RcJson);
+    expect(deriveBudgetsFromLhrSpy).not.toHaveBeenCalled();
+    expect(writeFileSpy).toHaveBeenCalled();
+  });
+
+  it('should create derived budgets if generateBudgets is true and lhr is passed', async () => {
+    jest.spyOn(fileHelpers, 'readFile')
+      .mockReturnValueOnce('' as never)
+      .mockReturnValueOnce('{}' as never);
```

Comment:
OOS note: In the future, I recommend simplifying this test. If I were to debug/modify this test as someone who did not write this feature, I would need to investigate to understand the function call order instead of treating the function as a black box with inputs and outputs.

Reply:
Fully agree but in the future i would like to have a function called check if exist and not use readFile with a return value of string to check if it exists. Maybe then this would not look confusing